sample_apps/simple_app_python/example_app.py

```diff
@@ -277,28 +270,71 @@ def server_dispatch(data_dir, server_app_host, server_app_port):
     return True
 
 ###############################################################################
-def client_dispatch(data_dir, server_app_host, server_app_port):
-    """
-    Start a client app that sends a message via secure SSL connection.
+def setup_server_ssl_context(cctm, data_dir:str, print_all:bool):
     """
-    print(fnl(), 'Start client application process ...\n')
+    Setup a SSL context for server-process, verifying certificates v/s
+    root certificate issued by CA.
 
-    # Client needs to authenticate the server; hence 'Purpose.SERVER_AUTH'
-    context = ssl.create_default_context(ssl.Purpose.SERVER_AUTH)
+    **** NOTE: **** SSL interfaces need to read the private key from a file.
+
+    So, this method invokes a Certifier trust manager's method to persist
+    the private-key data to a named temporary file. This is recognized as
+    a potential security lapse, where private-key info is available,
+    albeit very briefly, in a temp-file.
+
+    However, the window of abuse is somewhat mitigated as:
+
+      - Name of the tmp-dir and tmp-file is randomly generated
+      - The window-of-exposure is small; between dump_private_key_to_file()
+        and context.load_cert_chain(), below.
+      - The tmp-dir/tmp-file is wiped out immediately after SSL certification,
+        by context.load_cert_chain(), succeeds
+
+    This is the best we can do right now with existing Python SSL interfaces.
+    """
+    # Server needs to authenticate the client; hence 'Purpose.CLIENT_AUTH'
+    context = ssl.create_default_context(ssl.Purpose.CLIENT_AUTH)
     context.verify_mode = ssl.CERT_REQUIRED
 
     cert_outfilename = os.path.join(data_dir, APPLN_CERT_SIGNED_BY_ROOT_CA)
-    key_outfilename  = os.path.join(data_dir, APPLN_KEYF_SIGNED_BY_ROOT_CA)
-    context.load_cert_chain(certfile = cert_outfilename, keyfile = key_outfilename)
+
+    with TemporaryDirectory() as tempdir:
+        with NamedTemporaryFile('wt', dir=tempdir, delete=False) as temp_keyf:
+            temp_keyf.close()
```

Comment:
why do we have to close the file here?

Reply:
Actually you are right, this is not really needed. Took it out and the test still works.